### PR TITLE
WIP: Closes #330: Singed messages

### LINKED ISSUE #330: Sign typed data messages
**Is your feature request related to a problem? Please describe.**
The contract allows you to represent contract signatures. Currently it is not possible, while using the apps, to just sign payloads.

**Describe the solution you'd like**
The application should be able to sign payloads provided by web dApps. In order to do that, the app should handle requests from the browser extension to sign those payloads so that they can be authenticated by the dApp.

This will be done in two different stages:

- [x] Mobile app receives push notification to sign a payload and returns a single signature (App signature) - see https://github.com/gnosis/safe-android/pull/338
- [ ] Mobile app receives an intent to sign a specific payload and waits for other owners of the safe to sign the payload and then builds the final signature  

**Notes**
For more information see epic: https://github.com/gnosis/safe/issues/78

### PR DESCRIPTION
**Pull request by rmeissner**
Closes #330.

Changes proposed in this pull request:
- TDB

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/res/layout/view_confirm_resend_panel.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="utf-8"?>
+<merge
```
Comment (rmeissner) [inline app/src/main/res/layout/view_confirm_resend_panel.xml]: merge doesn't really make sense here, right? a normal include would work just as well.

**Comment by rmeissner** [comment]
Some of the comments might relate to existing / old code and can be ignored for now.

File: app/src/main/res/layout/view_confirm_resend_panel.xml (hunk 1 above)
Comment (elgatovital) [reply app/src/main/res/layout/view_confirm_resend_panel.xml]: yes, actually both tags are unnecessary - just having a root view group is enough

File: app/src/main/java/pm/gnosis/heimdall/helpers/MessageSignatureStore.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import pm.gnosis.svalinn.accounts.base.models.Signature
+import java.util.concurrent.CopyOnWriteArraySet
+import javax.inject.Inject
+
+interface MessageSignatureStore {
+    fun store(signature: Signature)
+    fun observe(): Observable<Set<Signature>>
+}
+
+class DefaultMessageSignatureStore @Inject constructor() : ObservableOnSubscribe<Set<Signature>>, MessageSignatureStore {
```
[the remark this answers is not in the document]
Comment (elgatovital) [reply app/src/main/java/pm/gnosis/heimdall/helpers/MessageSignatureStore.kt]: yes, using SignatureStore now

**Comment by jpalvarezl** [comment]
This pull request should have the new safe_v1 branch as the target it should merge into.

**Comment by rmeissner** [comment]
not relevant anymore

**Pull request closed without merging**